File: model/src/main/kotlin/net/dungeonhub/model/score/ScoreUpdateModel.java
package net.dungeonhub.model.score;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import me.taubsie.dungeonhub.common.DungeonHubService;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ScoreUpdateModel {
    private long id;
    private long amount;

    public String toJson() {
        return DungeonHubService.getInstance()
                .getGson()
                .toJson(this);
    }
}

File: model/src/main/kotlin/net/dungeonhub/model/score/ScoreUpdateModel.kt
package net.dungeonhub.model.score

import net.dungeonhub.service.MoshiService
import net.dungeonhub.structure.model.UpdateModel

class ScoreUpdateModel(
    val id: Long,
    var amount: Long
) : UpdateModel<ScoreModel> {
    fun toJson(): String {
        return MoshiService.moshi.adapter(ScoreUpdateModel::class.java).toJson(this)
    }
}